File: models/fitnessgoal.py
# models/fitnessgoal.py
from sqlalchemy import Column, Integer, String, Date, ForeignKey
from sqlalchemy.orm import relationship
from .base import Base

class FitnessGoal(Base):
    __tablename__ = "fitnessgoal"  # matches FitnessGoal table (lowercased)

    goal_id = Column(Integer, primary_key=True, autoincrement=True)
    member_id = Column(Integer, ForeignKey("member.member_id", ondelete="SET NULL"))
    goal_type = Column(String, nullable=False)
    value = Column(Integer)
    deadline = Column(Date, nullable=False)

    member = relationship("Member", back_populates="goals")

File: health_club_management/src/main/java/healthClubManagement/fitnessGoal.java
package healthClubManagement;

import jakarta.persistence.Entity;

@Entity
@Table(Name = "FitnessGoal")

public class FitnessGoal {
    package healthClubManagement;

import jakarta.persistence.*;
import java.time.LocalDate;

    @Entity
    @Table(name = "FitnessGoal")
    public class FitnessGoal {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "goal_id")
        private Long goalId;

        @ManyToOne
        @JoinColumn(name = "member_id")
        private Member member;

        @Column(name = "goal_type", nullable = false)
        private String goalType;

        @Column(name = "value")
        private Integer value;

        @Column(name = "deadline", nullable = false)
        private LocalDate deadline;

        public Long getGoalId() {
            return goalId;
        }

        public void setGoalId(Long goalId) {
            this.goalId = goalId;
        }

        public Member getMember() {
            return member;
        }

        public void setMember(Member member) {
            this.member = member;
        }

        public String getGoalType() {
            return goalType;
        }

        public void setGoalType(String goalType) {
            this.goalType = goalType;
        }

        public Integer getValue() {
            return value;
        }

        public void setValue(Integer value) {
            this.value = value;
        }

        public LocalDate getDeadline() {
            return deadline;
        }

        public void setDeadline(LocalDate deadline) {
            this.deadline = deadline;
        }

        @Override
        public String toString() {
            return "FitnessGoal{" +
                    "goalId=" + goalId +
                    ", member=" + (member != null ? member.getMemberId() : null) +
                    ", goalType='" + goalType + '\'' +
                    ", value=" + value +
                    ", deadline=" + deadline +
                    '}';
        }
    }

}